backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java

```diff
@@ -2369,34 +2556,36 @@ void testEndpointTest() {
         final InternalEndpointTestRequest sentPayload = capturedPayload.getValue();
 
         Assertions.assertEquals(createdEndpoint.getId(), sentPayload.endpointUuid, "the sent endpoint UUID in the payload doesn't match the one from the fixture");
-        Assertions.assertEquals(orgId, sentPayload.orgId, "the sent org id in the payload doesn't match the one from the fixture");
+        Assertions.assertEquals(DEFAULT_ORG_ID, sentPayload.orgId, "the sent org id in the payload doesn't match the one from the fixture");
         assertNull(sentPayload.message, "the sent message should be null since no custom message was specified");
     }
 
     /**
      * Tests that when the "test endpoint" handler is called with an endpoint
      * UUID that doesn't exist, a not found response is returned.
      */
-    @Test
-    void testEndpointTestNotFound() {
-        final String accountId = "test-endpoint-test-account-number";
-        final String orgId = "test-endpoint-test-org-id";
+    @ParameterizedTest
+    @ValueSource(booleans = {false, true})
+    void testEndpointTestNotFound(final boolean isKesselEnabled) {
+        this.kesselTestHelper.mockKesselRelations(isKesselEnabled);
 
-        final String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(accountId, orgId, "user-name");
+        final String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(DEFAULT_ACCOUNT_ID, DEFAULT_ORG_ID, DEFAULT_USER);
         final Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
 
         MockServerConfig.addMockRbacAccess(identityHeaderValue, FULL_ACCESS);
 
         // Call the endpoint under test.
-        final String path = String.format("/endpoints/%s/test", UUID.randomUUID());
+        final UUID randomId = UUID.randomUUID();
+        this.kesselTestHelper.mockKesselPermission(DEFAULT_USER, IntegrationPermission.TEST, ResourceType.INTEGRATION, randomId.toString());
```

Comment:
I agree, yes. I have modified the code so that we return `404`s when the endpoints do not exist... the problem I see is that many functions fetch the endpoint at a later stage, so we end up doing kind of the following:

```java
public void handler(@Param final UUID endpointId) {
    if (!endpointRepository.existsById(endpointId)) {
        throw new NotFoundException();
    }

     // ... If Kessel this, if RBAC that and then...
    innerFunction();
}

public void innerFunction(final UUID endpointId) {
    // ...
    final Endpoint dbEndpoint = endpointRepository.getById(endpointId);
    if (dbEndpoint == null) {
        throw new NotFoundException();
    }
    // ...
}
```

So we have two database calls for basically the same thing. So I wonder if I should extract all those database calls and move them to the initial handlers instead, and then pass the fetched endpoint directly to the inner functions :thinking: ... But maybe that should be part of a different PR, what do you think?

Reply:
Ok, I think we can keep it simple as it is then.